### ISSUE TITLE
Download RHS ligand structures

### ISSUE THREAD

**Issue by mwinokan** [m2ms/fragalysis-frontend#1527]
follows on from #1203 

@mwinokan to complete spec

**Comment by mwinokan** [comment]
@kaliif has started to work on this, it may be that creating a new custom SDF from the metadata may be trickier (would need a new ticket)

**Comment by kaliif** [comment]
@mwinokan, the CSV is generated in the frontend,  so the missing values must be populated there. Unless we think this is a poor solution and move the CSV generation to backend as well.

For the second point, custom pdbs, I need some test data.

**Comment by mwinokan** [comment]
@kaliif for the custom PDB's can you see if this works for you:

[A71EV2A_lhs2rhs_20241010_capitalprefix.zip](https://github.com/user-attachments/files/17376435/A71EV2A_lhs2rhs_20241010_capitalprefix.zip)

**Comment by mwinokan** [comment]
@kaliif has made some progress locally and is progressing with the test data for custom PDBs. After this @kaliif can work on adding the metadata.csv information to the SDF output

@boriskovar-m2ms please show @matej-vavrek where to find the f/e code to add the extra columns to the metadata.csv

**Comment by mwinokan** [comment]
First two bullets need review in staging by @mwinokan

@matej-vavrek will work on the updated CSV columns now

**Comment by matej-vavrek** [comment]
@mwinokan, there are probably two things mentioned here. I am sorry raising this so late.
Download RHS csv file served by F/E
![obrázok](https://github.com/user-attachments/assets/2118ba47-743e-49e6-8669-194108549def)
Download structures containing metadata.csv served by B/E
![obrázok](https://github.com/user-attachments/assets/6f3214e0-21e5-4d0e-93bd-46a70346358a)
Which one needs to be updated, please?

Also I noticed in download structures that F/E is getting full link for download instead of only a part (file_url=..). Will it be like this and we need to update download link, @kaliif?
`http://127.0.0.1:8080/api/download_structures/?file_url=http://127.0.0.1:8080/viewer/react/download/tag/c6988237-606f-48c5-a3fb-892e87373c77`

**Comment by mwinokan** [comment]
No worries @matej-vavrek. It's the CSVs of the RHS compound sets and selected compounds that need to be changed please

**Comment by kaliif** [comment]
> Also I noticed in download structures that F/E is getting full link for download instead of only a part (file_url=..). Will it be like this and we need to update download link, @kaliif? `http://127.0.0.1:8080/api/download_structures/?file_url=http://127.0.0.1:8080/viewer/react/download/tag/c6988237-606f-48c5-a3fb-892e87373c77`

@matej-vavrek where is this link coming from?

**Comment by matej-vavrek** [comment]
@kaliif if I am looking correctly it comes from `/api/download_structures` which returns `file_url`

**Comment by matej-vavrek** [comment]
@kaliif regarding the third point, we use `text_scores` from `/api/compound-mols-scores/` but they are all `nan` there. Could it be used or?
![obrázok](https://github.com/user-attachments/assets/80641356-4c1f-4445-829e-453dca7762f1)

**Comment by kaliif** [comment]
@matej-vavrek these values come from the uploaded sdf, the backend doesn't modify them.

**Comment by Waztom** [comment]
Validation step should pick these up, they need to be populated with non-nan values, and feedback if  "nan" values found to uploader. 

@mwinokan to confirm what .sdf was used to test and if there were in fact "nan" values for the scores.

**Comment by mwinokan** [comment]
@tdudgeon says that the validator should be changed for the following fields:

- ref_url
- method
- submitter_name
- submitter_institution
- submitter_email

@kaliif please verify or implement the following

- The header molecule must have values for all those fields (and validated e.g. email should be an email)
- Ignore values for all those fields in non-header molecules
- Populate the `text_scores` values for all the compounds from the header molecule

**Comment by kaliif** [comment]
>  ![obrázok](https://private-user-images.githubusercontent.com/60509086/379725680-80641356-4c1f-4445-829e-453dca7762f1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzE1MDE0ODAsIm5iZiI6MTczMTUwMTE4MCwicGF0aCI6Ii82MDUwOTA4Ni8zNzk3MjU2ODAtODA2NDEzNTYtNGMxZi00NDQ1LTgyOWUtNDUzZGNhNzc2MmYxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEzVDEyMzMwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBhY2E4ODRlMTdkNDU5YjNiNTdkMGIzYzI0NzJjM2VjZjgxN2YyYjYyZjM3NGMyYWVlZmNlMGE0MGJhZTBiNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QuSSMIriaZ0elLSaf24zYeERg6TYYTWnZqLko2Eca94)

@mwinokan there's a field `ganeration_date` that you have not listed but is `"nan"`. Is that allowed to be `nan` (or `None`) or should it be included in validation?

**Comment by mwinokan** [comment]
@kaliif Apologies I'd forgotten about that one. It should indeed be required and date-like and not null/na/None.

**Comment by kaliif** [comment]
@mwinokan re validating date, does it always come in a certain format or does it need to accept different ones?

**Comment by mwinokan** [comment]
No, we are not currently prescribing a strict format. Perhaps you can use [dateutil](https://pypi.org/project/python-dateutil/)? But if not please specify what the best format for Django is

**Comment by mwinokan** [comment]
@kaliif right. Please support day-first in case of ambiguity

**Comment by mwinokan** [comment]
@kaliif says this is merged to staging

**Comment by mwinokan** [comment]
Created spinoff ticket #1583

**Comment by kaliif** [comment]
>    * Print a warning on upload for nan values in the SDF

@mwinokan,  apologies, missed that bit of the spec. 
At the moment though, there's no way to issue warnings, it can list errors on failure but otherwise, it's being treated as a success. I'd have to add a mechanism to collect and return them to the user without halting the upload. OK to do this under this ticket?
